src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java

```diff
@@ -627,16 +825,30 @@ private void doTestGetCaseByCaseRef(CaseType caseType, boolean caseEvents) throw
     CaseRequestDTO requestParams = new CaseRequestDTO(caseEvents);
     CaseDTO results = target.getCaseByCaseReference(testCaseRef, requestParams);
 
-    CaseDTO expectedCaseResult = createExpectedCaseDTO(caseFromCaseService, caseEvents);
+    List<DeliveryChannel> expectedAllowedDeliveryChannels = null;
+
+    if (caseFromCaseService.getCaseType().equals("SPG") && caseFromCaseService.isHandDelivery()) {
+      expectedAllowedDeliveryChannels = Arrays.asList(DeliveryChannel.SMS);
+    } else {
+      expectedAllowedDeliveryChannels = Arrays.asList(DeliveryChannel.POST, DeliveryChannel.SMS);
+    }
```

Comment:
I'm not sure that it can - this method is called by more than one test. The value of expectedAllowedDeliveryChannels gets set differently depending on what value of caseType is passed in by the calling test.

Reply:
now refactored as discussed